src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int AMOUNT = 4;
+    private static final int MAX_VALUE = 5;
+    private static final int MIN_VALUE = 5;
+    private static final int DEFAULT_RADIUS = 10;
+    private static final String DEFAULT_COLOR = Color.WHITE.name();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        if (getQuantityIndex() == 0) {
+            return getSquare();
+        } else if (getQuantityIndex() == 1) {
+            return getRightTriangle();
+        } else if (getQuantityIndex() == 2) {
+            return getIsoscelesTrapezoid();
+        } else if (getQuantityIndex() == 3) {
+            return getCircle();
+        } else {
+            return getRectangle();
+        }
+    }
+
+    private Rectangle getRectangle() {
+        Rectangle rectangle = new Rectangle(colorSupplier.getRandomColor(),
+                getRandomParameters(), getRandomParameters());
+        rectangle.drawFigure();
+        return rectangle;
+    }
+
+    private Circle getCircle() {
+        Circle circle = new Circle(colorSupplier.getRandomColor(), getRandomParameters());
+        circle.drawFigure();
+        return circle;
+    }
```

Comment:
```suggestion
    private Circle getRandomCircle() {
        return  new Circle(colorSupplier.getRandomColor(), getRandomParameters());
    }
```
Same for other methods